File: patterns/rule-strukturierung/rules-js/rule-strukturierung.js
rules.JSRule({
  name: "1-2-3 Rule Structure",
  triggers: [
    triggers.ItemStateChangeTrigger("Foo"),
    triggers.ItemStateChangeTrigger("Bar"),
    triggers.ItemStateChangeTrigger("Baz")
  ],
  execute: (event) => {
    const foo = items.getItem("Foo").state;
    const bar = items.getItem("Bar").state;
    const baz = items.getItem("Baz").state;

    // 1. Pruefen, ob Regel laufen muss
    if (foo === "NULL" || bar === "NULL" || baz === "NULL") {
      console.warn("One of the Items is NULL");
      return;
    }

    // 2. Berechnen
    const onCount = [foo, bar, baz].filter((s) => s === "ON").length;
    const newState = onCount >= 2 ? "ON" : "OFF";

    // 3. Ausfuehren
    items.getItem("Buzz").sendCommand(newState);
  }
});

File: patterns/rule-strukturierung/rules-python/rule_strukturierung.py
from openhab import rule, Registry
from openhab.triggers import ItemStateChangeTrigger


@rule(triggers=[
    ItemStateChangeTrigger("Foo"),
    ItemStateChangeTrigger("Bar"),
    ItemStateChangeTrigger("Baz"),
])
class OneToTwoThreeRuleStructure:
    def execute(self, module, input):
        foo = str(Registry.getItem("Foo").state)
        bar = str(Registry.getItem("Bar").state)
        baz = str(Registry.getItem("Baz").state)

        # 1. Pruefen, ob Regel laufen muss
        if "NULL" in (foo, bar, baz):
            self.logger.warn("One of the Items is NULL")
            return

        # 2. Berechnen
        on_count = [foo, bar, baz].count("ON")
        new_state = "ON" if on_count >= 2 else "OFF"

        # 3. Ausfuehren
        Registry.getItem("Buzz").sendCommand(new_state)
